### PR TITLE
Use Dataclass for ASE and QE

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

**Comment by samwaseda** [comment]
From my point of view, it makes more sense to have pre-defined fields in `AtomisticsOutput`, such as `energy`, `forces` etc., since we have already seen in `pyiron_atomistics` that by not defining fields, we could easily end up having different tags for the same values (cf. SPHInX & VASP)

**Comment by jan-janssen** [comment]
> From my point of view, it makes more sense to have pre-defined fields in `AtomisticsOutput`, such as `energy`, `forces` etc., since we have already seen in `pyiron_atomistics` that by not defining fields, we could easily end up having different tags for the same values (cf. SPHInX & VASP)

Done

File: atomistics/calculators/ase.py [hunk 1]
```diff
@@ -29,93 +52,40 @@ def evaluate_with_ase(
...
-            results["stress"] = calc_stress_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
+        quantities = []
+        if "calc_energy" in tasks:
+            quantities.append("energy")
+        if "calc_forces" in tasks:
+            quantities.append("forces")
+        if "calc_stress" in tasks:
+            quantities.append("stress")
```
Comment (samwaseda) [inline atomistics/calculators/ase.py:70]: Here I see the same problem as the one we discussed [here](https://github.com/pyiron/atomistics/pull/118) and [here](https://github.com/pyiron/atomistics/pull/121). To the very least, these lines appear twice inside atomistics, so they have to be somehow refactored, but more fundamentally, I see a clear connection between tags, tasks (i.e. `calc_XYZ`) and interactive getters. Interactive getters and the tags are now fully conceptualised with the data class, and I think the same should happen for the tasks as well.

Comment (liamhuber) [reply atomistics/calculators/ase.py:70]: 110%, although fair to break that step into a different PR

**Comment by liamhuber** [comment]
It's not mission critical because one can always refactor: rename later, but it might be wise to resolve #129 before/while propagating these classes to the other calculators

**Review by samwaseda — APPROVED** [review]
There's still no concept for the part that I mentioned, but it's now refactored, so I'm ok with the current state

**Comment by jan-janssen** [comment]
> There's still no concept for the part that I mentioned, but it's now refactored, so I'm ok with the current state

I agree I played around with it a bit and was not able to solve it, so I would prefer to move forward with the current pull request and then do the optimisation in a separate pull request.

**Pull request merged**